File: python/svix/api/streaming.py
# This file is @generated
from ..models import EndpointHeadersOut, HttpSinkHeadersPatchIn, SinkTransformationOut
from .common import ApiBase
from .streaming_event_type import (
    StreamingEventType,
    StreamingEventTypeAsync,
)
from .streaming_events import (
    StreamingEvents,
    StreamingEventsAsync,
)
from .streaming_sink import (
    StreamingSink,
    StreamingSinkAsync,
)
from .streaming_stream import (
    StreamingStream,
    StreamingStreamAsync,
)


class StreamingAsync(ApiBase):
    @property
    def event_type(self) -> StreamingEventTypeAsync:
        return StreamingEventTypeAsync(self._client)

    @property
    def events(self) -> StreamingEventsAsync:
        return StreamingEventsAsync(self._client)

    @property
    def sink(self) -> StreamingSinkAsync:
        return StreamingSinkAsync(self._client)

    @property
    def stream(self) -> StreamingStreamAsync:
        return StreamingStreamAsync(self._client)

    async def sink_headers_get(
        self, stream_id: str, sink_id: str
    ) -> EndpointHeadersOut:
        """Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return EndpointHeadersOut.model_validate(response.json())

    async def sink_headers_patch(
        self,
        stream_id: str,
        sink_id: str,
        http_sink_headers_patch_in: HttpSinkHeadersPatchIn,
    ) -> EndpointHeadersOut:
        """Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks."""
        response = await self._request_asyncio(
            method="patch",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
            json_body=http_sink_headers_patch_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return EndpointHeadersOut.model_validate(response.json())

    async def sink_transformation_get(
        self, stream_id: str, sink_id: str
    ) -> SinkTransformationOut:
        """Get the transformation code associated with this sink."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return SinkTransformationOut.model_validate(response.json())


class Streaming(ApiBase):
    @property
    def event_type(self) -> StreamingEventType:
        return StreamingEventType(self._client)

    @property
    def events(self) -> StreamingEvents:
        return StreamingEvents(self._client)

    @property
    def sink(self) -> StreamingSink:
        return StreamingSink(self._client)

    @property
    def stream(self) -> StreamingStream:
        return StreamingStream(self._client)

    def sink_headers_get(self, stream_id: str, sink_id: str) -> EndpointHeadersOut:
        """Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks."""
        response = self._request_sync(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return EndpointHeadersOut.model_validate(response.json())

    def sink_headers_patch(
        self,
        stream_id: str,
        sink_id: str,
        http_sink_headers_patch_in: HttpSinkHeadersPatchIn,
    ) -> EndpointHeadersOut:
        """Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks."""
        response = self._request_sync(
            method="patch",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/headers",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
            json_body=http_sink_headers_patch_in.model_dump_json(
                exclude_unset=True, by_alias=True
            ),
        )
        return EndpointHeadersOut.model_validate(response.json())

    def sink_transformation_get(
        self, stream_id: str, sink_id: str
    ) -> SinkTransformationOut:
        """Get the transformation code associated with this sink."""
        response = self._request_sync(
            method="get",
            path="/api/v1/stream/{stream_id}/sink/{sink_id}/transformation",
            path_params={
                "stream_id": stream_id,
                "sink_id": sink_id,
            },
        )
        return SinkTransformationOut.model_validate(response.json())

File: java/lib/src/main/java/com/svix/api/Streaming.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.EndpointHeadersOut;
import com.svix.models.HttpSinkHeadersPatchIn;
import com.svix.models.SinkTransformationOut;

import lombok.Getter;

import okhttp3.HttpUrl;

import java.io.IOException;

public class Streaming {
    private final SvixHttpClient client;

    @Getter private final StreamingEventType eventType;
    @Getter private final StreamingEvents events;
    @Getter private final StreamingSink sink;
    @Getter private final StreamingStream stream;

    public Streaming(SvixHttpClient client) {
        this.client = client;
        this.eventType = new StreamingEventType(client);
        this.events = new StreamingEvents(client);
        this.sink = new StreamingSink(client);
        this.stream = new StreamingStream(client);
    }

    /** Get the HTTP sink headers. Only valid for `http` or `otelTracing` sinks. */
    public EndpointHeadersOut sinkHeadersGet(final String streamId, final String sinkId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/headers", streamId, sinkId));
        return this.client.executeRequest("GET", url.build(), null, null, EndpointHeadersOut.class);
    }

    /** Updates the Sink's headers. Only valid for `http` or `otelTracing` sinks. */
    public EndpointHeadersOut sinkHeadersPatch(
            final String streamId,
            final String sinkId,
            final HttpSinkHeadersPatchIn httpSinkHeadersPatchIn)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/headers", streamId, sinkId));
        return this.client.executeRequest(
                "PATCH", url.build(), null, httpSinkHeadersPatchIn, EndpointHeadersOut.class);
    }

    /** Get the transformation code associated with this sink. */
    public SinkTransformationOut sinkTransformationGet(final String streamId, final String sinkId)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(
                                String.format(
                                        "/api/v1/stream/%s/sink/%s/transformation",
                                        streamId, sinkId));
        return this.client.executeRequest(
                "GET", url.build(), null, null, SinkTransformationOut.class);
    }
}
